src/main/java/Duke.java

```diff
@@ -1,10 +1,378 @@
+                for(int i = 1; i< count; i++){
+                    System.out.println(i + ". " + type[i] + marks[i] + " " + tasks[i]);
+                }
+                System.out.println(line);
+            }
+
+            else if (text.contains("mark")){
+                System.out.println(line);
+                String strNum;
+                int num;
+                if(text.contains("unmark")){ //Unmark
+                    System.out.println("OK, I've marked this task as not done yet:");
+                    strNum = text.substring(7, text.length());
+                    num = Integer.parseInt(strNum);
+                    marks[num] = "[ ]";
+                    System.out.println(marks[num] + " " + tasks[num]);
+
+                }
+                else{ //Mark
+                    System.out.println("Nice! I've marked this task as done:");
+                    strNum = text.substring(5, text.length());
+                    num = Integer.parseInt(strNum);
+                    marks[num] = "[X]";
+                    System.out.println(marks[num] + " " + tasks[num]);
+                }
+
+                System.out.println(line);
+            }
+
+            else if (text.contains("todo")) {
+                if(text.equals("todo") || text.equals("todo ")){
+                    System.out.println(line);
+                    System.out.println("☹ OOPS!!! The description of a todo cannot be empty.");
+                    System.out.println(line);
+                } else{
+                    System.out.println(line);
+                    System.out.println("Got it. I've added this task");
+                    text = text.substring(5, text.length());
+                    System.out.println("  [T][ ] " + text);
+                    System.out.println("Now you have " + count + " tasks in the list.");
+                    System.out.println(line);
+                    tasks[count] = text;
+                    type[count] = "[T]";
+                    marks[count] = "[ ]";
+                    count++;
+                }
+                
+            }
+
+            else if (text.contains("deadline")) {
+                System.out.println(line);
+                System.out.println("Got it. I've added this task");
+
+                // Modify text according to deadline
+                text = text.substring(9, text.length());
+                int indexOfSlash = text.indexOf("/",9);
+                String day = text.substring(indexOfSlash+4, text.length());
+                String task = text.substring(0, indexOfSlash);
+                text = task + "(by: " + day + ")";
+
+                System.out.println("  [D][ ] " + text);
+                System.out.println("Now you have " + count + " tasks in the list.");
+                System.out.println(line);
+                tasks[count] = text;
+                type[count] = "[D]";
+                marks[count] = "[ ]";
+                count++;
```

Comment:
Consider extracting the details into other methods (i.e. SLAP more) to make the code more readable.